__tests__/index.js

```diff
@@ -1,71 +1,18 @@
-import forEach from 'callbag-for-each'
-import map from 'callbag-map'
 import pipe from 'callbag-pipe'
-import subject from 'callbag-subject'
-import take from 'callbag-take'
+import { listenable, pullable, willBe } from '@jeetiss/callbag-marble-tester'
 
 import concatMap from '../src'
 
-test('works', () => {
-  const actual = []
-
-  let source = subject()
-  let inner
-
+test('works with listenable sources', () =>
   pipe(
-    source,
-    concatMap(value => {
-      inner = subject()
-      return pipe(
-        inner,
-        map(innerValue => `${value}: ${innerValue}`),
-        take(value + 2),
-      )
-    }),
-    forEach(data => {
-      actual.push(data)
-    }),
-  )
+    listenable('--o----o-|'),
```

Comment:
ok, im seriously starting to love this thing :P can final subscription be controlled with a marble diagram? we'll need it to test when sources get unsubscribed

Reply:
yep, now I compare the order of events and unsubscribe between diagram in `willBe` sink and final source, or did you mean something else?

Reply:
I'd like to be able to test a scenario like this https://github.com/ReactiveX/rxjs/blob/33058d3ba33151a306b2d2e8ff6106e3796edfce/spec/operators/concatMap-spec.ts#L504

at the moment (from what I understand) `willBe` creates infinite subscription, but it would be good to test a scenario when `willBe` unsubscribes when i.e. inner is still active, or when there is no current inner, but outer is still active

Reply:
Yes, I'll see how it was done in rxjs